sdk/storage/inc/adls/protocol/adls_rest_client.hpp

```diff
@@ -0,0 +1,2220 @@
+  enum class PathLeaseAction
+  {
+    Acquire,
+    Break,
+    Change,
+    Renew,
+    Release,
+    Unknown
+  };
+
+  static std::string PathLeaseActionToString(const PathLeaseAction& pathLeaseAction)
+  {
+    switch (pathLeaseAction)
+    {
+      case PathLeaseAction::Acquire:
+        return "acquire";
+      case PathLeaseAction::Break:
+        return "break";
+      case PathLeaseAction::Change:
+        return "change";
+      case PathLeaseAction::Renew:
+        return "renew";
+      case PathLeaseAction::Release:
+        return "release";
+      default:
+        return std::string();
+    }
+  }
+
+  static PathLeaseAction PathLeaseActionFromString(const std::string& pathLeaseAction)
+  {
+    if (pathLeaseAction == "acquire")
+    {
+      return PathLeaseAction::Acquire;
+    }
+    if (pathLeaseAction == "break")
+    {
+      return PathLeaseAction::Break;
+    }
+    if (pathLeaseAction == "change")
+    {
+      return PathLeaseAction::Change;
+    }
+    if (pathLeaseAction == "renew")
+    {
+      return PathLeaseAction::Renew;
+    }
+    if (pathLeaseAction == "release")
+    {
+      return PathLeaseAction::Release;
+    }
+    throw std::runtime_error("Cannot convert " + pathLeaseAction + " to PathLeaseAction");
+  };  // enum class PathLeaseAction
+
+  // Optional. If the value is "getStatus" only the system defined properties for the path are returned. If the value is "getAccessControl" the access control list is returned in the response headers (Hierarchical Namespace must be enabled for the account), otherwise the properties are returned.
+  enum class PathGetPropertiesAction
+  {
+    GetAccessControl,
+    GetStatus,
+    Unknown
+  };
+
+  static std::string PathGetPropertiesActionToString(const PathGetPropertiesAction& pathGetPropertiesAction)
+  {
+    switch (pathGetPropertiesAction)
+    {
+      case PathGetPropertiesAction::GetAccessControl:
+        return "getAccessControl";
+      case PathGetPropertiesAction::GetStatus:
+        return "getStatus";
+      default:
+        return std::string();
+    }
+  }
+
+  static PathGetPropertiesAction PathGetPropertiesActionFromString(const std::string& pathGetPropertiesAction)
+  {
+    if (pathGetPropertiesAction == "getAccessControl")
+    {
+      return PathGetPropertiesAction::GetAccessControl;
+    }
+    if (pathGetPropertiesAction == "getStatus")
+    {
+      return PathGetPropertiesAction::GetStatus;
+    }
+    throw std::runtime_error("Cannot convert " + pathGetPropertiesAction + " to PathGetPropertiesAction");
+  };  // enum class PathGetPropertiesAction
+
+  struct ServiceListFileSystemsOptions
+  {
+    std::string Prefix = std::string();  // Filters results to filesystems within the specified prefix.
```

Comment:
any reason we need to initialize string in this struct with `= std::string()` while not need to do so in other structs?

Reply:
Optional parameters are deliberately declared this and initialized this way in`XXOptions`. All mandatory parameters are not.
Parameters are not initialized for `XXResponse` structs.

Reply:
i'm not sure i can agree with you.

`std::string Prefix = std::string()` and `std::string Prefix` have exactly the same effect. But the former looks more complicated and superfluous.

Also, I think it's very hard for people to realize (at least I cannot) that this variable is optional just by adding `= std::string()`.

Reply:
Please note that this is only for the developer of Azure Storage's convenience layer to note that those variables are optional and are not exposed to other customers. So I would counter-argue that at least when I tried to code the convenience APIs, I won't have to look at the REST API to actually know what to put in options and what to put directly in API as a mandatory parameter.
Another counter-argument on it not being superfluous is that this is a reflection of a tag defined in swagger spec, which has no better ways of being shown in the auto-generated code. I considered adding `Required` or `Optional` suffix to the member name but that looks weirder than this.